handlers/libhoney_event_handler.go

```diff
@@ -86,40 +86,53 @@ func initLibhoney(config config.Config, version string) func() {
 	return libhoney.Close
 }
 
+// setTimestampsAndDurationIfValid sets time-related fields in the emitted telemetry
+// about the request/response cycle.
+//
+// It only sets timestamps if they are present in the captured event, and only
+// computes and includes durations for which there are correct timestamps to based them upon.
+func setTimestampsAndDurationIfValid(honeyEvent *libhoney.Event, httpEvent assemblers.HttpEvent) {
+	honeyEvent.AddField("meta.httpEvent_handled_at", time.Now())
+	switch true {
+	case httpEvent.RequestTimestamp.IsZero() && httpEvent.ResponseTimestamp.IsZero():
+		// no request or response, which is weird, but let's send what we do know
+		honeyEvent.AddField("meta.timestamps_missing", "request, response")
+		honeyEvent.Timestamp = time.Now()
+
+	case httpEvent.RequestTimestamp.IsZero():
+		// no request
+		honeyEvent.AddField("meta.timestamps_missing", "request")
+		// but we have a response
+		honeyEvent.Timestamp = httpEvent.ResponseTimestamp
+		honeyEvent.AddField("http.response.timestamp", httpEvent.ResponseTimestamp)
+		honeyEvent.AddField("meta.httpEvent_response_handled_latency_ms", time.Since(httpEvent.ResponseTimestamp).Milliseconds())
+
+	case httpEvent.ResponseTimestamp.IsZero(): // have request, no response
+		// no response
+		honeyEvent.AddField("meta.timestamps_missing", "response")
+		// but we have a request
+		honeyEvent.Timestamp = httpEvent.RequestTimestamp
+		honeyEvent.AddField("http.request.timestamp", httpEvent.RequestTimestamp)
+		honeyEvent.AddField("meta.httpEvent_request_handled_latency_ms", time.Since(httpEvent.RequestTimestamp).Milliseconds())
+
+	default: // the happiest of paths, we have both request and response
+		honeyEvent.Timestamp = httpEvent.RequestTimestamp
+		honeyEvent.AddField("http.request.timestamp", httpEvent.RequestTimestamp)
+		honeyEvent.AddField("http.response.timestamp", httpEvent.ResponseTimestamp)
+		honeyEvent.AddField("meta.httpEvent_request_handled_latency_ms", time.Since(httpEvent.RequestTimestamp).Milliseconds())
+		honeyEvent.AddField("meta.httpEvent_response_handled_latency_ms", time.Since(httpEvent.ResponseTimestamp).Milliseconds())
+		honeyEvent.AddField("duration_ms", httpEvent.ResponseTimestamp.Sub(httpEvent.RequestTimestamp).Milliseconds())
+	}
+}
+
+// handleEvent transforms a captured httpEvent into a libhoney event and sends it
 func (handler *libhoneyEventHandler) handleEvent(event assemblers.HttpEvent) {
-	// create libhoney event
-	ev := libhoney.NewEvent()
+	// the telemetry event to send
+	var ev *libhoney.Event = libhoney.NewEvent()
```

Comment:
```suggestion
	ev = libhoney.NewEvent()
```

Reply:
Sooo, I threw `var …` in there purely a trick to help the human reading this code later to differentiate between all the things we've got that are variously event-flavored. I like this feature and propose that we keep it, but it's not a hill I'll even be maimed much less die on. I recognize that this is not a popular idiom, but what even _are_ idioms, anyway?

VS Code and [the Go extension](https://marketplace.visualstudio.com/items?itemName=golang.go) currently displays the comment above an explicit variable declaration in a floaty box that appears whenever the mouse hovers over the variable.

![Screenshot 2023-10-03 at 12 08 10 PM](https://github.com/honeycombio/honeycomb-network-agent/assets/517302/67ded239-4803-4c05-a4b5-7a67aa39b81a)

An implicit variable declaration does not trigger the Go extension to include a comment (either above or inline) for the variable.

![Screenshot 2023-10-03 at 12 07 44 PM](https://github.com/honeycombio/honeycomb-network-agent/assets/517302/117cbac4-4de8-4b13-b2cd-b6ac9c3b04a8)